api4/cloud.go

```diff
@@ -409,7 +409,25 @@ func sendAdminUpgradeRequestEmail(c *Context, w http.ResponseWriter, r *http.Req
 		return
 	}
 
-	if err = c.App.SendAdminUpgradeRequestEmail(user.Username, sub); err != nil {
+	body, e := ioutil.ReadAll(r.Body)
+	if err != nil {
+		c.Err = model.NewAppError("Api4.sendAdminUpgradeRequestEmail", "api.cloud.app_error", nil, e.Error(), http.StatusInternalServerError)
+		return
+	}
+	defer r.Body.Close()
+
+	var upgReqT *model.FreeTierUpgradeRequestType
+	if e = json.Unmarshal(body, &upgReqT); err != nil {
+		c.Err = model.NewAppError("Api4.sendAdminUpgradeRequestEmail", "api.cloud.app_error", nil, e.Error(), http.StatusInternalServerError)
+		return
+	}
```

Comment:
The current endpoint `/cloud/subscription/limitreached/invite` it's explicit that is for when a user is trying to `invite` others when the limit is reached. 

I believe the intent was to have another one under that handler like:

`/cloud/subscription/limitreached/join` for the case that someone is trying to join when the limit is reached. This way we can simplify here the need for this extra payload and other checks below.

Reply:
I have updated the code